base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -110,9 +127,9 @@ pub fn verify_header(header: &BlockHeader) -> Result<MoneroPowData, MergeMineErr
 }
 
 /// Extracts the Monero block hash from the coinbase transaction's extra field
-pub fn extract_tari_hash(monero: &monero::Block) -> Result<Option<monero::Hash>, MergeMineError> {
-    let extra_field = ExtraField::try_parse(&monero.miner_tx.prefix.extra)
-        .map_err(|_| MergeMineError::DeserializeError("Invalid extra field".to_string()))?;
+pub fn proxy_extract_tari_hash_from_block(monero: &monero::Block) -> Result<Option<monero::Hash>, MergeMineError> {
+    // When we extract the merge mining hash, we do not care if the extra field can be parsed without error.
+    let extra_field = parse_extra_field_truncate_on_error(&monero.miner_tx.prefix.extra);
```

Comment:
DNM! This will allow different proof of work data byte vecs to have the same difficulty

Reply:
Dont think this is possible: the function will only extract the tari hash from the already mine block. The Monero code keeps the extra field as a vec<u8> only only deserializes it if used by the code base. So all the bytes are mined, if they are correct and deserializable or not is a different question. But they are all locked by mining

Reply:
Correct @SWvheerden